src/libraries/System.Text.Json/src/System/Text/Json/BitStack.cs

```diff
@@ -129,6 +129,8 @@ private bool PopFromArray()
             return (_array[elementIndex] & (1 << extraBits)) != 0;
         }
 
+        public readonly bool Peek() => _currentDepth <= AllocationFreeMaxDepth ? (_allocationFreeContainer & 1) != 0 : PeekInArray();
```

Comment:
Just by looking at their names, it isn't entire clear to me how `Peek` differs from `PeekInArray`? Perhaps a bit of renaming and/or using comments might help.

Reply:
added comments